### PR TITLE
Remove GitSharp dependency

### PR DESCRIPTION
**Pull request by qinezh**
@vicancy @vwxyzh @superyyrrzz @ansyral @hellosnow @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.RestApi/SwaggerModelConverter.cs [hunk 1]
```diff
@@ -97,7 +97,12 @@ public static RestApiRootItemViewModel FromSwaggerModel(SwaggerModel swagger)
                             };
 
                             // TODO: line number
-                            itemVm.Metadata[Constants.PropertyName.Source] = swagger.Metadata[Constants.PropertyName.Source];
+                            object value;
+                            if (swagger.Metadata.TryGetValue(Constants.PropertyName.Source, out value))
+                            {
+                                itemVm.Metadata[Constants.PropertyName.Source] = value;
+                            }
+                            itemVm.Metadata[Constants.PropertyName.Source] = null;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.RestApi/SwaggerModelConverter.cs]: >itemVm.Metadata[Constants.PropertyName.Source] = null; [](start = 28, length = 54)

put it in else clause?

File: src/Microsoft.DocAsCode.Common/CommandUtility.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+{
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public static class CommandUtility
+    {
+        public static int RunCommand(CommandInfo commandInfo, StreamWriter stdoutwWriter = null, StreamWriter stderrWriter = null, int timeoutInMilliseconds = Timeout.Infinite)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Common/CommandUtility.cs]: >w [](start = 81, length = 1)

remove

**Comment by superyyrrzz** [comment]
After this change, git becomes another prerequisite of DocFX. Should update document and build.cmd later?

File: src/docfx/SubCommands/BuildCommand.cs [hunk 3]
```diff
@@ -275,9 +277,26 @@ private static void MergeOptionsToConfig(BuildCommandOptions options, BuildJsonC
             config.GlobalMetadata = GetGlobalMetadataFromOption(config.GlobalMetadata, options.GlobalMetadataFilePath, config.GlobalMetadataFilePaths, options.GlobalMetadata);
         }
 
+        private static void InitialEnvrionmentContext(BuildJsonConfig config)
```
Comment (superyyrrzz) [inline src/docfx/SubCommands/BuildCommand.cs]: >Initial [](start = 28, length = 7)

=> Initialize?

**Comment by qinezh** [comment]
Yes, I'll update document in next PR, but I think there's no need to update build.cmd, since we don't require git command when we build the source code of docfx.

---
In reply to: [261479285](https://github.com/dotnet/docfx/pull/909#issuecomment-261479285) [](ancestors = 261479285)

**Comment by superyyrrzz** [comment]
:shipit:

File: src/docfx/SubCommands/BuildCommand.cs [hunk 4]
```diff
@@ -275,9 +277,26 @@ private static void MergeOptionsToConfig(BuildCommandOptions options, BuildJsonC
             config.GlobalMetadata = GetGlobalMetadataFromOption(config.GlobalMetadata, options.GlobalMetadataFilePath, config.GlobalMetadataFilePaths, options.GlobalMetadata);
         }
 
+        private static void InitializeEnvrionmentContext(BuildJsonConfig config)
```
Comment (ansyral) [inline src/docfx/SubCommands/BuildCommand.cs]: >InitializeEnvrionmentContext [](start = 28, length = 28)

typo

**Comment by vicancy** [comment]
:shipit:

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**